src/main/kotlin/org/rust/ide/annotator/RsExternalLinterPass.kt

```diff
@@ -144,8 +147,29 @@ class RsExternalLinterPass(
         }
     }
 
+    /**
+     * We pass suppress actions as options directly to [HighlightInfo.registerFix] because otherwise
+     * actions from [IntentionManager.getStandardIntentionOptions] will be added to the options list
+     */
     private val highlights: List<HighlightInfo>
-        get() = annotationHolder.map(HighlightInfo::fromAnnotation)
+        get() = annotationHolder.map { annotation ->
+            val quickFixes = annotation.quickFixes?.toList().orEmpty()
+            annotation.quickFixes?.clear()
+            val highlight = HighlightInfo.fromAnnotation(annotation)
+            val element = file.findElementAt(highlight.startOffset) ?: file.findElementAt(highlight.endOffset - 1)
+            val problemGroup = highlight.problemGroup as? SuppressableProblemGroup
+            val options = problemGroup?.getSuppressActions(element)?.toList()
+            for (quickFix in quickFixes) {
+                highlight.registerFix(
+                    quickFix.quickFix,
+                    options,
+                    HighlightDisplayKey.getDisplayNameByKey(quickFix.key),
+                    quickFix.textRange,
+                    quickFix.key
+                )
+            }
+            highlight
+        }
```

Comment:
Looks quite hacky
Maybe `IntentionActionWithOptions` may help here to provide custom options?

Reply:
> Maybe `IntentionActionWithOptions` may help here to provide custom options?

The options from `IntentionActionWithOptions.getOptions` are used only if `key`  in `HighlightInfo.getOptions` is `null`. But to make it `null` we need to register the fix with `null` instead of the actual or default key, so we need most of the code above anyway.